pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularUpdateConsentCopyEvents.java

```diff
@@ -0,0 +1,123 @@
+import org.broadinstitute.ddp.model.event.CopyAnswerEventAction;
+import org.broadinstitute.ddp.studybuilder.ActivityBuilder;
+import org.broadinstitute.ddp.studybuilder.EventBuilder;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.customizer.BindList;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Task to fix for Singular Consent copy events to handle parent/self name overriding.
+ */
+@Slf4j
+public class SingularUpdateConsentCopyEvents implements CustomTask {
+    private static final String ACTIVITY_DATA_FILE = "patches/singular-consent-copy-events.conf";
+
+    private Path cfgPath;
+    private Config studyCfg;
+    private Config varsCfg;
+    private Config activityDataCfg;
+
+    private Handle handle;
+    private StudyDto studyDto;
+    private UserDto adminUser;
+
+    @Override
+    public void init(Path cfgPath, Config studyCfg, Config varsCfg) {
+        this.cfgPath = cfgPath;
+        this.studyCfg = studyCfg;
+        this.varsCfg = varsCfg;
+
+        File file = cfgPath.getParent().resolve(ACTIVITY_DATA_FILE).toFile();
+        if (!file.exists()) {
+            throw new DDPException("Data file is missing: " + file);
+        }
+        this.activityDataCfg = ConfigFactory.parseFile(file).resolveWith(varsCfg);
+    }
+
+    @Override
+    public void run(Handle handle) {
+        this.studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(studyCfg.getString("study.guid"));
+        this.adminUser = handle.attach(JdbiUser.class).findByUserGuid(studyCfg.getString("adminUser.guid"));
+        this.handle = handle;
+
+        fixConsentCopyEvent();
+    }
+
+    private void fixConsentCopyEvent() {
+        String activityCode = "CONSENT_PARENTAL";
+        log.info("Working on copy event for {}...", activityCode);
+        long activityId = ActivityBuilder.findActivityId(handle, studyDto.getId(), activityCode);
+        long copyConfigId = handle.attach(EventDao.class)
+                .getAllEventConfigurationsByStudyId(studyDto.getId())
+                .stream()
+                .filter(event -> event.getEventActionType().equals(EventActionType.COPY_ANSWER))
+                .filter(event -> {
+                    if (event.getEventTriggerType().equals(EventTriggerType.ACTIVITY_STATUS) && event.getPreconditionExpression() == null) {
+                        var trigger = (ActivityStatusChangeTrigger) event.getEventTrigger();
+                        return trigger.getStudyActivityId() == activityId
+                                && trigger.getInstanceStatusType() == InstanceStatusType.COMPLETE;
+                    }
+                    return false;
+                })
+                .map(event -> ((CopyAnswerEventAction) event.getEventAction()).getCopyConfigurationId())
```

Comment:
To avoid NPE I'd suggest using two sequential maps:
```
.map(EventDto::getEventAction)
.map(EventAction::getCopyConfigurationId)
```

Reply:
One time use patch code with specific expectation that events exist.